helm-chart/README.md

```diff
@@ -0,0 +1,636 @@
+- Helm version 3.2.0 or later
+- Dynamic volume provisioning support in the underlying infrastructure (for PostgreSQL persistence)
+- Familiarity with Kubernetes concepts and Helm charts
+
+## Chart Components
+
+This Helm chart deploys the following components:
+
+1. **Formbricks Application**: The core Formbricks service.
+2. **PostgreSQL Database**: (Optional) A relational database for Formbricks data.
+3. **Redis**: (Optional) An in-memory data structure store for caching.
+4. **Traefik Ingress Controller**: (Optional) A modern HTTP reverse proxy and load balancer.
+
+## Installation
+
+### Quick Start
+
+To quickly deploy Formbricks with default settings:
+
+1. Add the Formbricks Helm repository:
+   ```bash
+   helm repo add formbricks https://charts.formbricks.com
+   helm repo update
+   ```
+
+2. Install the chart:
+   ```bash
+   helm install my-formbricks formbricks/formbricks --namespace formbricks --create-namespace
+   ```
+
+This will deploy Formbricks with default settings, including a new PostgreSQL instance and Redis enabled, Traefik disabled.
+
+### Usage Examples
+
+Here are various examples of how to install and configure the Formbricks Helm chart:
+
+1. **Default Installation with Traefik enabled and Custom Hostname**:
+
+    <details>
+
+    <summary>Option 1: Installation without SSL (Not recommended for production)</summary>
+
+      ```bash
+      helm install my-formbricks formbricks/formbricks \
+        --namespace formbricks \
+        --create-namespace \
+        --set traefik.enabled=true \
+        --set hostname=forms.example.com
+      ```
+
+      This command enables Traefik and sets a custom hostname. Replace `forms.example.com` with your actual domain name.
+
+    </details>
+
+    <details>
+    <summary>Option 2: Installation with SSL (Recommended for production)</summary>
+
+    1. First, download the values file:
+
+    ```bash
+    helm show values formbricks/formbricks > values.yaml
+    ```
+
+    2. Open the `values.yaml` file in a text editor and make the following changes:
+
+    ```yaml
+    traefik:
+      enabled: true
+      additionalArguments:
+        - "--certificatesresolvers.letsencrypt.acme.email=your-email@example.com"
+    ```
+
+    Replace `your-email@example.com` with a valid email address where you want to receive Let's Encrypt notifications.
+
+    3. Install Formbricks with the updated values file:
+
+    ```bash
+    helm install my-formbricks formbricks/formbricks \
+      -f values.yaml \
+      --namespace formbricks \
+      --create-namespace \
+      --set hostname=forms.example.com
+    ```
+
+    This command enables Traefik, sets a custom hostname, and uses the configured email address for Let's Encrypt. Remember to replace `forms.example.com` with your actual domain name.
+
+    </details>
+
+These installation options provide flexibility in setting up Formbricks with Traefik. The SSL option is recommended for production environments to ensure secure communications.
```

Comment:
I have an issue running this on a digital ocean k8s cluster. I first deployed the helm chart to the k8s cluster on digital ocean using this command:

```
helm install formbricks . \
  -f values.yaml \
  --namespace formbricks \
  --create-namespace \
  --set hostname=k8s.formbricks.com
```

This also created a new load balancer on digital ocean, I waited for the load balancer to be created, got the public IP address and create the DNS record.

Unfortunately, this seems to take too long, as in the meantime traefik tried to create the certificate, but failed because the domain wasn't set up at that time:

```
2024-10-04T14:00:09Z ERR github.com/traefik/traefik/v3/pkg/provider/acme/provider.go:396 > Unable to obtain ACME certificate for domains error="unable to generate a certificate for the domains [k8s.formbricks.com]: error: one or more domains had a problem:\n[k8s.formbricks.com] acme: error: 400 :: urn:ietf:params:acme:error:dns :: DNS problem: NXDOMAIN looking up A for k8s.formbricks.com - check that a DNS record exists for this domain; DNS problem: NXDOMAIN looking up AAAA for k8s.formbricks.com - check that a DNS record exists for this domain\n" ACME CA=https://acme-v02.api.letsencrypt.org/directory acmeCA=https://acme-v02.api.letsencrypt.org/directory domains=["k8s.formbricks.com"] providerName=letsencrypt.acme routerName=formbricks-formbricks-e30908acec5bc4dc11a4@kubernetescrd rule=Host(⁠ k8s.formbricks.com ⁠)
```

It seems like we need a way to manually say that traefik should now begin with creating the certificate after the load balancer is created and user set up the DNS records.
How did you manage to get it working in your testing with digital ocean?

Reply:
The logic is the same for all known K8s providers like EKS and AKS, and providing public Load Balancer IPs generally takes some time. In a scenario where the definitions are done correctly, SSL will be generated in subsequent checks anyway.